components/datePicker/datePickerContainer.tsx

```diff
@@ -38,6 +38,7 @@ export const DatePickerContainer: FC<DatePickerContainerProps> = ({ error, label
         className={'bg-dark'}
         containerClassName={s.cont}
         dateSeparator={' - '}
+        format={router.locale === 'en' ? 'DD/MM/YYYY' : 'MM.DD.YYYY'}
```

Comment:
This is an excellent and elegant solution. However, it is also inflexible and crude because it does not allow the code to scale in the future when new locales are added. In addition, the `DD/MM/YYYY` date format does not accurately represent the `en` locale. Its true value is `MM/DD/YYYY`, which can be calculated automatically by the application environment.

A solution already exists to define the date format in a user locale using `Intl.DateTimeFormat`.
As a variant [working example](https://stackoverflow.com/questions/43368659/how-to-determine-users-locale-date-format-using-javascript-format-is-dd-mm-or-m).

Since you are using the `useRouter` hook, you can create your own custom `useDateFormat` hook that will return the required format. You can also specify `options` in the `Intl.DateTimeFormat` constructor, forcing the two-digit day and month format and the full year format. [Read the documentation for an explanation.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat#locale_options)

Reply:
i fixed it with hook **useDatePickerFormat**.
```javascript 
export const useDatePickerFormat = () => {
  const router = useRouter();

  const localeFormat = router.locale === 'en' ? 'MM/DD/YYYY' : 'DD.MM.YYYY';

  return { localeFormat };
};